DataDog/datadog-agent
# Split package scripts by package type

### What does this PR do?

Splits the package scripts that run during the Agent installation in a single script per package type, instead of having a single script and trying to detect which distro we are on.

### Motivation

Removes lots of conditional logic, in exchange for some code duplication (code has grown by ~300 lines).
Makes the scripts work on all distros and not only the ones we can detect.

### Describe how to test/QA your changes

Run an installation, upgrade and uninstall on each OS family:
- debian
- rhel
- suse
- macos (except for uninstall)

And each Agent flavor:
- dogstatsd
- iot-agent
- agent

How to verify the scripts ran:
* Make sure the `dd-agent` user is created on install and deleted on uninstall when uninstalling with `apt remove --purge`. Note: for RPM packages we don't delete it and you will have to delete it yourself after each uninstall with `userdel dd-agent`.
* For non-dogstatsd, also verify the `install_info` is created in `/etc/datadog-agent` (`/opt/datadog-agent/etc/` on macos) and deleted on uninstall.
